lib/Site.js

```diff
@@ -185,6 +188,39 @@ Site.prototype.collectBaseUrl = function () {
   return Promise.resolve();
 };
 
+/**
+ * Collects the user defined variables map in the site/subsites
+ */
+Site.prototype.collectUserDefinedVariablesMap = function () {
+  // The key is the base directory of the site/subsites,
+  // while the value is user defined variable map for that particular subsite
```

Comment:
"// while the value is `a mapping of` user defined variable`s`~` map for that particular subsite`~"